packages/util-validation/lib/core/Validator.mjs

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-unused-vars */
+import {NotYetImplementedError} from '@zorko-io/util-error'
+
+/**
+ * Validator, parse passed parameters, can fill defaults and
+ * raise errors if params not feet to validation rules.
+ * It's an agnostic to validation rules, so subclasses can
+ * choose what validation engine/rules to use  for declaration
+ */
+
+export class Validator {
+  /**
+   * Asynchronously parse and validate input params, fill with defaults if needed
+   * @param {Object} params - input value to validate
+   * @return {Promise<ValidationOutput>} - params and error if any
+   */
+
+  async parse(params) {
```

Comment:
why do you change `validate` to `parse`?

Reply:
for 2 reasons:

1. during validation we also fill with defaults, so we change an input, it's kind of transformation or parsing initial structure and providing it's copy + modification as output
1.  having `validate` inside `Validator` looks too verbose, for example `validator.validate(...)`